sources/shaders/Stride.Core.Shaders/Ast/Hlsl/ByteAddressBufferType.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Stride.Core.Shaders.Ast.Hlsl
+{
+    public static class ByteAddressBufferType
+    {
+        public static readonly ObjectType ByteAddressBuffer = new ObjectType("ByteAddressBuffer");
+
+        public static readonly ObjectType RWByteAddressBuffer = new ObjectType("RWByteAddressBuffer");
+
+        private static readonly ObjectType[] ObjectTypes = new[] { ByteAddressBuffer, RWByteAddressBuffer };
+
+        public static bool IsByteAddressBufferType(this TypeBase type)
+        {
+            return Parse(type.Name) != null;
+        }
+
+        /// <summary>
+        /// Parses the specified name.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <returns></returns>
+        public static ObjectType Parse(string name)
+        {
+            foreach (var objectType in ObjectTypes)
+            {
+                if (string.Compare(name, objectType.Name.Text, StringComparison.OrdinalIgnoreCase) == 0)
+                    return objectType;
+            }
+            return null;
+        }
```

Comment:
This function operates like a .Net ``TryParse``, ``Parse``'s common operating behavior in .Net is to throw when the input not valid.
Maybe change the function signature to ``public static bool TryParse(string name, out ObjectType type)``

Reply:
Yes I made it like this as it matches the other `Parse(string)` methods in the shader, see `SamplerType`. I prefer to keep it consistent but could change if needed, if so should I update the other places as well or might that be fit for a separate PR?

Reply:
Makes sense, I'd also go with consistency. Such a fix could then be provided by a separate PR for all `Parse` methods in the same way.

Reply:
I see, yes, let's keep it that way for now